lib/ansible/modules/cloud/amazon/guardduty_ip_list.py

```diff
@@ -0,0 +1,235 @@
+    choices: ['TXT', 'STIX', 'OTX_CSV', 'ALIEN_VAULT', 'PROOF_POINT', 'FIRE_EYE']
+  location:
+    description:
+    - The URI of the file that contains the IP list.
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+
+EXAMPLES = r'''
+- name: create GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: true
+
+- name: disable GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: false
+
+- name: delete GuardDuty detector
+  guardduty_threatlist:
+    state: absent
+'''
+
+
+RETURN = r'''#'''
+
+import os
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def ip_list_exists(client, module, result):
+    try:
+        if module.params.get('list_type') == 'trusted':
+            sets = client.list_ip_sets(
+                DetectorId=module.params.get('detector_id')
+            )
+            for i in sets['IpSetIds']:
+                ti_set = client.get_ip_set(
+                    DetectorId=module.params.get('detector_id'),
+                    IpSetId=i
+                )
+                if ti_set['Name'] == module.params.get('name'):
+                    result['trusted_set_id'] = i
+                    return True
+        if module.params.get('list_type') == 'threat':
+            sets = client.list_threat_intel_sets(
+                DetectorId=module.params.get('detector_id')
+            )
+            for i in sets['ThreatIntelSetIds']:
+                ti_set = client.get_threat_intel_set(
+                    DetectorId=module.params.get('detector_id'),
+                    ThreatIntelSetId=i
+                )
+                if ti_set['Name'] == module.params.get('name'):
+                    result['threat_set_id'] = i
+                    return True
+    except ClientError:
+        return False
+
+    return False
+
+
+def create_ip_list(client, module, params, result):
+    if module.check_mode:
+        module.exit_json(changed=True)
+    try:
+        if module.params.get('list_type') == 'trusted':
+            response = client.create_ip_set(**params)
+            result['trusted_set_id'] = response['IpSetId']
+            result['changed'] = True
+            return result
+        if module.params.get('list_type') == 'threat':
+            response = client.create_threat_intel_set(**params)
+            result['threat_set_id'] = response['ThreatIntelSetId']
+            result['changed'] = True
+            return result
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to create IP list")
+
+    return result
+
+
+def update_ip_list(client, module, params, result):
```

Comment:
It looks like this doesn't need `result` as a parameter and could just take a set_id instead. result['changed'] = True could be set where this is called since that's always the case for this code path. It doesn't look like this function needs to return anything.